### ISSUE TITLE
Broken subscriptions in 0.7.0

### ISSUE THREAD

**Issue by alx13** [Urigo/graphql-modules#431]
I was trying to get Subscriptions working in 0.7.0, but unfortunately this Promise is never resolved:
https://github.com/Urigo/graphql-modules/blob/master/packages/core/src/graphql-module.ts#L586

So when client is trying to subscribe as
```graphql
subscription {
  ping 
}
``` 
TestProvider is not instantiated and `onConnect` hook is never called on backend. (not getting console.log neither `'on construct'` nor  `'on connect'`)

Version `0.6.6` works fine. 

Code to reproduce
```typescript
import 'reflect-metadata';
import {
  GraphQLModule,
  ModuleContext,
  OnConnect,
  OnRequest,
} from '@graphql-modules/core';
import {
  Injectable,
  ProviderScope,
} from '@graphql-modules/di';
import {
  gql,
  PubSub,
  ApolloServer,
} from 'apollo-server-express';
import express = require('express');
import { createServer } from 'http';

@Injectable({
  scope: ProviderScope.Session,
})
class TestProvider implements OnConnect, OnRequest {
  public constructor() {
    console.log('on construct');
  }

  public async onConnect() {
    console.log('on connect');
  }

  public async onRequest() {
    console.log('on request');
  }
}

export const AppModule = new GraphQLModule({
  name: 'App',
  typeDefs: gql`
    type Query {
      ping: Boolean!
    }
    type Mutation {
      pong: Boolean!
    }
    type Subscription {
      ping: Boolean!
    }
  `,
  resolvers: {
    Query: {
      ping() {
        console.log('ping');

        return true;
      },
    },
    Mutation: {
      async pong(_root: any, _args: any, { injector }: ModuleContext) {
        await injector.get(PubSub).publish('ping', { ping: false });

        return false;
      },
    },
    Subscription: {
      ping: {
        subscribe(_root: any, _args: any, { injector }: ModuleContext) {
          console.log('on subscribe');

          return injector.get(PubSub).asyncIterator('ping');
        },
      },
    },
  },
  providers: [
    PubSub,
    TestProvider,
  ],
});

const { schema, subscriptions, context } = AppModule;


const PORT = 4000;

const app = express();

const apollo = new ApolloServer({ schema, context, subscriptions });

apollo.applyMiddleware({
  app,
  path: '/graphql',
});

const ws = createServer(app);

apollo.installSubscriptionHandlers(ws);

ws.listen(PORT, () => {
  console.log(`Apollo Server is now running on http://localhost:${PORT}/graphql`);
});
```

versions:
```yaml
dependencies:
  '@graphql-modules/core': 0.7.0
  '@graphql-modules/di': 0.7.0
  apollo-server-express: 2.4.8
  express: 4.16.4
  graphql: 14.2.1
  reflect-metadata: 0.1.13
  tslib: 1.9.3
devDependencies:
  '@types/express': 4.16.1
  '@types/node': 10.14.4
  ts-node: 8.1.0
  typescript: 3.4.3
```

**Comment by ardatan** [comment]
Hi @alx13 
Could you try with these canary versions;
```
 - @graphql-modules/core@0.8.0-alpha.eb196efa
 - @graphql-modules/di@0.8.0-alpha.eb196efa
```

**Comment by alx13** [comment]
Hi @ardatan it's working in canary :)

**Comment by ardatan** [comment]
Fixed in 0.7.1
